src/hooks/activityNavigation/ActivityProvider.tsx

```diff
@@ -9,33 +9,44 @@ interface ActivityProviderProps {
   children?: React.ReactNode
   activities: Array<Activity>
 }
-
+/**
+ * Provider to store ID of the current activity being shown to user
+ * and to determine the next activity from the activities list.
+ */
 export const ActivityProvider: FC<ActivityProviderProps> = ({
   children,
   activities,
 }) => {
-  const [currentActivity, setCurrentActivity] = useState(0)
+  const [currentActivityId, setCurrentActivityId] = useState<string>('')
 
-  const handleSetCurrent = () => {
-    if (
-      activities.length === 0 ||
-      activities[currentActivity].status === ActivityStatus.Active
-    ) {
-      return
-    }
-    const currentActivityId = activities[currentActivity].id
+  useEffect(() => {
+    findNextActiveActivity()
+  }, [])
 
-    const nextActivityIndex = activities.findIndex(
-      (activity) =>
-        activity.status === ActivityStatus.Active &&
-        activity.id !== currentActivityId
+  const findCurrentActivity = (): Activity | undefined => {
+    return activities.find(({ id }) => id === currentActivityId)
+  }
+
+  const findNextActiveActivity = (): Activity | undefined => {
+    return activities.find(
+      ({ status, id }) =>
+        status === ActivityStatus.Active && id !== currentActivityId
     )
+  }
 
-    if (nextActivityIndex === -1 || currentActivity === nextActivityIndex) {
+  const handleSetCurrent = () => {
+    const currentActivity = findCurrentActivity()
+
+    // if current activity is still active, then do not move
+    // to the next activity unless it is marked as completed
+    if (currentActivity?.status === ActivityStatus.Active) {
```

Comment:
I can think of two (broad) reasons why currentActivity will be undefined:
1. currentActivityId has not been set yet (is ``) so cannot be found in activities array
2. activities array is empty so currentActivityId cannot be found

in both cases we are doing work even though we don't need to, so we can check if `currentActivityId` is an empty string or activities array is empty, and return early

this logic can live in the `findCurrentActivity` function itself so it's handled silently, and here in `handleSetCurrent` we can explicitly handle if `currentActivity` is undefined.